### ISSUE TITLE
Keeping a Substructure Preserved During Mols Generation

### ISSUE THREAD

**Issue by OmarAlAttraqDev** [AustinT/mol_ga#9]
Hi,

Is it possible to keep a certain substructure (or atom_ids) of the starting mol intact (unchanged) during the generation process? If not, can you please provide me guide lines on how this might be implemented?

Thanks

**Comment by AustinT** [comment]
Hi @OmarAlAttraqDev, thank you for using mol_ga!

There is no option for this in mol_ga. If you want this, I think you would need to do two things:

1. Ensure that the desired substructure is present in at least one molecule of the starting population (via [this argument](https://github.com/AustinT/mol_ga/blob/cf92b96f6a5252b2efa7e1982740628d26135666/mol_ga/preconfigured_gas.py#L11))
2. Create an "offspring generation function" which preserves the substructure (via [this argument](https://github.com/AustinT/mol_ga/blob/cf92b96f6a5252b2efa7e1982740628d26135666/mol_ga/preconfigured_gas.py#L15))

This is the function used to generate offspring by default: <https://github.com/AustinT/mol_ga/blob/cf92b96f6a5252b2efa7e1982740628d26135666/mol_ga/graph_ga/gen_candidates.py#L64>

You could either replace this with your own function (accepting the same arguments), or wrap this function with another function which rejects molecules that do not contain your desired substructure. Does that make sense?

**Comment by OmarAlAttraqDev** [comment]
@AustinT Thank you very much for your help, it was really useful.
I added a substruct argument and passed it to the mutate / crossover funcs. I added a simple substructure check at the mutate and crossover in the check for new mols generated from running the reaction, such as [https://github.com/AustinT/mol_ga/blob/cf92b96f6a5252b2efa7e1982740628d26135666/mol_ga/graph_ga/mutate.py#L147C1-L148C1](url) , when checking the mol is ok, I also check the mol has the substructure present. 
I thought this way I avoid wasting cycles if the check was performed after a mol is selected from the reaction results. Currently, it seems to work as expected and the substructure is preserved.

Thanks
